pages/index.js

```diff
@@ -57,20 +57,164 @@ const ItemCard = styled.div`
   const [categories] = useState(initialCategories);
+  const [newItem, setNewItem] = useState({
+    name: "",
+    quantity: 1,
+    category: "",
+    comment: "",
+  });
+  const [errors, setErrors] = useState({});
+
+  useEffect(() => {
+    const storedItems = localStorage.getItem("shoppingItems");
+    if (storedItems) {
+      setItems(JSON.parse(storedItems));
+    } else {
+      setItems(
+        initialItems.map((item) => ({
+          ...item,
+          category: item.category.toLowerCase(),
+        }))
+      );
+    }
+  }, []);
+
+  useEffect(() => {
+    if (items.length > 0) {
+      localStorage.setItem("shoppingItems", JSON.stringify(items));
+    }
+  }, [items]);
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setNewItem({ ...newItem, [name]: value });
+  };
+
+  const validateForm = () => {
+    const errors = {};
+    if (!newItem.name) errors.name = "Name is required";
+    if (!newItem.quantity || newItem.quantity <= 0)
+      errors.quantity = "Quantity must be a positive number";
+    if (!newItem.category || newItem.category === "default")
+      errors.category = "Please select a category";
+    return errors;
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const formErrors = validateForm();
+    if (Object.keys(formErrors).length > 0) {
+      setErrors(formErrors);
+    } else {
+      setErrors({});
+      const updatedItems = [
+        { ...newItem, id: new Date().getTime().toString() },
+        ...items,
+      ];
+      setItems(updatedItems);
+      setNewItem({ name: "", quantity: 1, category: "", comment: "" });
+    }
+  };
 
   const totalItems = items.reduce((total, item) => total + item.quantity, 0);
 
   return (
     <StyledShoppingList>
       <h1>Shopping List: What I Need to Buy</h1>
+
+      <FormContainer onSubmit={handleSubmit}>
+        <h2>Add New Shopping Item</h2>
+        <div className="error">{errors.name}</div>
+        <label htmlFor="name">Item Name</label>
+        <input
+          type="text"
+          id="name"
+          name="name"
+          value={newItem.name}
+          onChange={handleInputChange}
+        />
+
+        <div className="error">{errors.quantity}</div>
+        <label htmlFor="quantity">Quantity</label>
+        <input
+          type="number"
+          id="quantity"
+          name="quantity"
+          value={newItem.quantity}
+          onChange={handleInputChange}
+        />
+
+        <div className="error">{errors.category}</div>
+        <label htmlFor="category">Category</label>
+        <select
+          id="category"
+          name="category"
+          value={newItem.category}
+          onChange={handleInputChange}
+        >
+          <option value="default">Please select a category</option>
+          {Object.keys(categories).map((category) => (
+            <option key={category} value={category}>
+              {categories[category]}
+            </option>
+          ))}
```

Comment:
Since categories is a static array like ["Dairy", "Bakery"], there is no need to use Object.keys(categories). You can directly map over the array, as this will be more efficient and clearer.